### PR TITLE
Store the output length in redis

### PR DESCRIPTION
**Pull request by dmathieu**
So we can truncate the content and still have the length incremented, to properly discard the content on a new request.

### CODE REVIEW

File: broker/io.go [hunk 1]
```diff
@@ -283,8 +284,12 @@ func Len(wd io.WriteCloser) (int64, error) {
 	conn := redisPool.Get()
 	defer conn.Close()
 
-	strlen, err := redis.Int64(conn.Do("STRLEN", w.channel.id()))
+	strlen, err := redis.Int64(conn.Do("GET", w.channel.lengthID()))
```
Comment (freeformzSFDC) [inline broker/io.go]: What happens when the key referred to by `lengthID()` isn't found?

**Comment by jabrown85** [comment]
Good catch @bensymonds!

File: broker/io.go (hunk 1 above)
Comment (dmathieu) [reply broker/io.go]: We get a `redis.ErrNil`, which is handled below.

Comment (dmathieu) [reply broker/io.go]: That's also already tested here: https://github.com/heroku/busl/blob/master/broker/io_test.go#L223

Comment (freeformzSFDC) [reply broker/io.go]: Why not fall back to STRLEN on the channel id?

Comment (freeformzSFDC) [reply broker/io.go]: FWIW: adding some code comments here that describe why things work the way they do would help future / new readers (doesn't have to be part of this PR of course).

Comment (dmathieu) [reply broker/io.go]: I've added a method comment: 5a42568f26057e266dedcd13d3cf172931927a57

**Comment by dmathieu** [comment]
You're right, the streams could have their output duplicated, as busl thinks there's nothing to discard. I've split this into 2 PRs.
This one now only stores the data. #213 uses it.

File: broker/io.go [hunk 2]
```diff
@@ -285,6 +289,10 @@ func Len(wd io.WriteCloser) (int64, error) {
 
 	strlen, err := redis.Int64(conn.Do("STRLEN", w.channel.id()))
 	if err != nil {
+		if err == redis.ErrNil {
+			return 0, nil
+		}
+
```
Comment (bensymonds) [inline broker/io.go]: Can you move this into #213 too? It's not required here.

Comment (dmathieu) [reply broker/io.go]: Sure. I didn't do it because it doesn't matter. But it can go into the other commit too.

File: broker/io.go [hunk 3]
```diff
@@ -275,6 +276,9 @@ func RenewExpiry(rd io.Reader) {
 }
 
 // Len returns the length of data already send to the reader
+// The length is stored in a separate key as opposed to using `STRLEN`, as the
+// content can be truncated if it reaches the size limit. When that happens, we
+// still want to increment the length though.
```
Comment (bensymonds) [inline broker/io.go]: Ugh sorry I should have this needs moving too.

**Pull request merged**